app/models/manageiq/providers/amazon/inventory_collection_default/cloud_manager.rb

```diff
@@ -176,6 +176,30 @@ def vm_and_template_labels(extra_attributes = {})
       attributes.merge!(extra_attributes)
     end
 
+    def vm_and_template_taggings(extra_attributes = {})
+      # TODO: make a generic Taggings IC and move it to base class?
+      attributes = {
+        :model_class                  => Tagging,
+        :association                  => :vm_and_template_taggings,
+        :manager_ref                  => [:taggable, :tag],
+        :parent_inventory_collections => [:vms, :miq_templates],
+        :inventory_object_attributes  => [
+          :taggable,
+          :tag,
+        ]
+      }
+
+      attributes[:targeted_arel] = lambda do |inventory_collection|
+        ids = inventory_collection.parent_inventory_collections.collect_concat(&:id)
```

Comment:
I was trying to "simplify" this to something like I did containers and avoid going through Vm.vm_and_template_taggings relation, just because it scared me.

But `add_inventory_collection` requires `:association => :foos` to be set, this determines the name of the `persister.foos` accessor to the collection :-(
Anyway, now that I added that relation in https://github.com/ManageIQ/manageiq/pull/16734 I should probably do what vm_and_template_labels does...